src/api/index.ts

```diff
@@ -20,17 +20,37 @@ export const fetchPokemon = async (id: number): Promise<PokemonData> => {
   const flavorTextEnglishObj = flavorTextEntries.find((obj: FlavorTextEntry) => obj.language?.name === 'en');
 
   let flavorText =  flavorTextEnglishObj ? flavorTextEnglishObj.flavor_text : '';
-  if (flavorText.toLowerCase().includes(name)) {
-    flavorText = flavorText.replace(name.toUpperCase(), `${flavorText.indexOf(name.toUpperCase()) ? 'i' : 'I'}t`);
-
-    // This handles some weird edge cases.
-    const properName = name[0].toUpperCase() + name.substr(1);
-    flavorText = flavorText.replace(properName, `${flavorText.indexOf(properName) ? 'i' : 'I'}t`);
-  }
+  flavorText = replacePokemonName(flavorText, name);
 
   return {
     name,
     image,
     flavorText,
   };
 }
+
+const replacePokemonName = (text: string, name: string) => {
+  if (text.toLowerCase().includes(name)) {
+    const properName = name[0].toUpperCase() + name.substr(1);
+    const words = text.toLowerCase().split(' ');
+
+    let wordIndex = words.findIndex((word) => word === name.toUpperCase());
+    let previousWord = words[wordIndex - 1];
+    if (previousWord === 'a' || previousWord === 'this') {
+      previousWord = `${text.indexOf(name.toUpperCase()) ? 't' : 'T'}his`
+      words[wordIndex] = 'Pokémon'
+    }
+
+    wordIndex = words.findIndex((word) => word === properName);
```

Comment:
The items within the `words` array are all lowercase (line 35). Since `properName` has the first letter capitalized (line 34), the `wordIndex` will always be `-1`.